libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -1202,6 +1203,7 @@ string ABIFunctions::abiDecodingFunctionCalldataArray(ArrayType const& _type)
 		Whiskers w{templ};
 		w("functionName", functionName);
 		w("readableTypeName", _type.toString(true));
+		// TODO this might make a difference
 		w("baseEncodedSize", toCompactHexWithPrefix(_type.isByteArray() ? 1 : _type.baseType()->calldataEncodedSize()));
```

Comment:
I have the feeling that ``baseEncodedSize`` should be 64 for an array with a base type of ``uint256[][2]`` - no test is failing here? Maybe we're lacking a suitable test for this...

Reply:
Or maybe not - I think I probably shouldn't claim half-thought-through worries after a quick glance and have a proper look once I'm back :-).